tests/test_lane.py

```diff
@@ -296,986 +291,1086 @@ def facetify(collection=Facets.COLLECTION_FULL,
+        # build its featured collection, but before it tries them it
+        # tries to build a collection based on items that are featured
+        # _within the CustomLists_.
+        lane.list_datasource = DataSource.lookup(
+            self._db, DataSource.GUTENBERG
         )
-        w, mw = _assert_expectations(
-            all_best_sellers, 2, 
-            lambda x: x.sort_title in (
-                self.fiction.sort_title, self.nonfiction.sort_title
+        self._db.commit()
+        eq_(True, lane.uses_customlists)
+        additional = [(Facets.COLLECTION_FULL, Facets.AVAILABLE_NOW, True),
+                      (Facets.COLLECTION_FULL, Facets.AVAILABLE_ALL, True)]
+        eq_(additional + default_facets, 
+            list(lane.featured_collection_facets()))
+
+    def test_apply_custom_filters(self):
+
+        # Create some works that will or won't show up in various
+        # lanes.
+        childrens_fiction = self._work(
+            fiction=True, with_license_pool=True, 
+            audience=Classifier.AUDIENCE_CHILDREN
+        )
+        nonfiction = self._work(fiction=False, with_license_pool=True)
+        childrens_fiction.target_age = tuple_to_numericrange((8,8))
+        self.add_to_materialized_view([childrens_fiction, nonfiction])
+
+        def match_works(lane, works, featured=False):
+            """Verify that calling apply_custom_filters to the given
+            lane yields the given list of works.
+            """
+            from model import MaterializedWork
+            base_query = self._db.query(MaterializedWork).join(
+                LicensePool, MaterializedWork.license_pool_id==LicensePool.id
             )
-        )
+            query, distinct = lane.apply_custom_filters(
+                self._db, base_query, MaterializedWork, featured
+            )
+            results = query.all()
+            works = sorted([(x.id, x.sort_title) for x in works])
+            materialized_works = sorted(
+                [(x.works_id, x.sort_title) for x in results]
+            )
+            eq_(works, materialized_works)
+            return distinct
+
+        # A lane may show only titles that come from a specific license source.
+        gutenberg_only = self._lane()
+        gutenberg_only.license_datasource = DataSource.lookup(
+            self._db, DataSource.GUTENBERG
+        )
+
+        distinct = match_works(gutenberg_only, [nonfiction])
+        # No custom list is involved, so there's no need to make the query
+        # distinct.
+        eq_(False, distinct)
+
+        # A lane may show fiction, nonfiction, or both.
+        fiction_lane = self._lane()
+        fiction_lane.fiction = True
+        match_works(fiction_lane, [childrens_fiction])
+
+        nonfiction_lane = self._lane()
+        nonfiction_lane.fiction = False
+        match_works(nonfiction_lane, [nonfiction])
+
+        both_lane = self._lane()
+        bo
```

Comment:
This is where comment https://github.com/NYPL-Simplified/server_core/pull/683#discussion_r145752977 was supposed to be.

The line is `younger_ya.target_age = (12,13)` in case it moves again.

You create this lane but you don't test it like you do with the older_ya lane.

Reply:
It did move again :angry:

Reply:
It should be on line 1287 - github is off by about 100 lines for some reason.

Reply:
You're right, that test is just absent.